mod_wasm/modules/wasm/config.m4

```diff
@@ -103,15 +103,45 @@ AC_DEFUN([APACHE_CHECK_WASMRUNTIME],[
       fi
     fi
 
-    AC_MSG_CHECKING([for Wasm Runtime version >= 0.5.0])
+    AC_MSG_CHECKING([for libwasm_runtime is available])
     AC_LANG([C])
     AC_TRY_COMPILE(
-      [#include <stdint.h>
-       #include "wasm_runtime.h"],
+      [#include "wasm_runtime.h"],
       [wasm_runtime_init_module();],
       [AC_MSG_RESULT(OK)
        ac_cv_wasmruntime=yes],
-      [AC_MSG_RESULT(FAILED)])
+      [AC_MSG_RESULT(FAILED)
+       ac_cv_wasmruntime=no]
+    )
+
+    mod_wasm_version_major=`grep MOD_WASM_VERSION_MAJOR modules/wasm/mod_wasm.h | cut -d' ' -f3`
+    mod_wasm_version_minor=`grep MOD_WASM_VERSION_MINOR modules/wasm/mod_wasm.h | cut -d' ' -f3`
+    mod_wasm_version_patch=`grep MOD_WASM_VERSION_PATCH modules/wasm/mod_wasm.h | cut -d' ' -f3`
+    mod_wasm_version="$mod_wasm_version_major"."$mod_wasm_version_minor"."$mod_wasm_version_patch"
+    AC_MSG_CHECKING([for mod_wasm $mod_wasm_version compatibility])
+    AC_LANG([C])
+    AC_RUN_IFELSE(
+      [AC_LANG_PROGRAM(
+        [#include <stdio.h>
+         #include "wasm_runtime.h"],
+        [ printf("\n");
+          printf("\tmod_wasm version is $mod_wasm_version\n");
+          printf("\tlibwasm_runtime version is %s\n", WASM_RUNTIME_VERSION);
+         if ( $mod_wasm_version_major == WASM_RUNTIME_VERSION_MAJOR
+           && $mod_wasm_version_minor <= WASM_RUNTIME_VERSION_MINOR )
```

Comment:
I don't know what kind of changes you're planning for wasm_runtime, but wouldn't it be more convenient to say that the minor version changes are always both forward and backward compatible, instead of only backward?

Reply:
Actually, the idea is to follow the [SemVer](https://semver.org/#semantic-versioning-specification-semver) spec.